modin/config/envvars.py

```diff
@@ -67,6 +74,85 @@ def get_help(cls) -> str:
         return help
 
 
+class EnvWithSibilings(
+    EnvironmentVariable,
+    # we have to pass anything here in order to derive from 'EnvironmentVariable',
+    # this doesn't force child classes to have 'str' type, they actually can be any type
+    type=str,
+):
+    """Ensure values synchronization between sibling parameters."""
+
+    _update_sibling = True
+
+    @classmethod
+    def _sibling(cls) -> type["EnvWithSibilings"]:
+        """Return a sibling parameter."""
+        raise NotImplementedError()
+
+    @classmethod
+    def get(cls) -> Any:
+        """
+        Get parameter's value and ensure that it's equal to the sibling's value.
+
+        Returns
+        -------
+        Any
+        """
+        if cls._sibling()._value is _UNSET and cls._value is _UNSET:
+            old_v: Any
+            new_v: Any
+            try:
+                old_v = cls._sibling()._get_raw_from_config()
+            except KeyError:
+                # keeping `_UNSET`, will handle default value later
+                # when process both of the siblings
+                old_v = _UNSET
+            try:
+                new_v = cls._get_raw_from_config()
+            except KeyError:
+                # keeping `_UNSET`, will handle default value later
+                # when process both of the siblings
+                new_v = _UNSET
+            if old_v is not _UNSET and new_v is _UNSET:
+                if not _TYPE_PARAMS[cls.type].verify(old_v):
+                    raise ValueError(f"Unsupported raw value: {old_v}")
+                old_v = _TYPE_PARAMS[cls.type].decode(old_v)
+                cls._sibling()._value = old_v
+                cls._sibling()._value_source = ValueSource.GOT_FROM_CFG_SOURCE
+
+                cls._value = old_v
+                cls._value_source = ValueSource.GOT_FROM_CFG_SOURCE
+                return cls._value
+            res = super().get()
+            cls._sibling()._value = res
+            cls._sibling()._value_source = cls._value_source
+            return res
+        return super().get()
+
+    @classmethod
+    def put(cls, value: Any) -> None:
+        """
+        Set a new value to this parameter as well as to its sibling.
+
+        Parameters
+        ----------
+        value : Any
+        """
+        super().put(value)
+        # avoid getting into an infinite recursion
+        if cls._update_sibling:
+            cls._update_sibling = False
+            try:
+                with warnings.catch_warnings():
+                    # filter potential future warnings of the sibling
+                    warnings.filterwarnings("ignore", category=FutureWarning)
+                    cls._sibling().put(value)
+            except BaseException:
```

Comment:
Let's put a comment when we get into except branch.

Reply:
just realized that this logic is incorrect, so removed the `except` block at all